server/models/Product.coffee

```diff
@@ -2,4 +2,14 @@ mongoose = require('mongoose')
 config = require '../../server_config'
 ProductSchema = new mongoose.Schema({}, {strict: false,read:config.mongo.readpref})
 
+ProductSchema.index({name: 1}, {name: 'name index'})
+
+ProductSchema.statics.findBasicSubscriptionForUser = (user) ->
+  basicProductName = 'basic_subscription'
+  if country = user.get 'country'
+    countrySpecificProductName = "#{country}_basic_subscription"
+    if countrySpecificProduct = yield @findOne {name: countrySpecificProductName}
+      return countrySpecificProduct
+  return yield @findOne {name: basicProductName}
```

Comment:
This can be wrapped. To do that:

```
co = require 'co'
ProductSchema.statics.findBasicSubscriptionForUser = co.wrap (user) ->
```

This makes it so that yield works, and when you call `findBasicSubscriptionForUser`, it returns a Promise which resolves or rejects the async return value appropriately.